fenix/app/src/main/java/org/mozilla/fenix/onboarding/HomeNotificationPermissionDialogFragment.kt

```diff
@@ -49,6 +53,7 @@ class HomeNotificationPermissionDialogFragment : DialogFragment() {
                         onDismiss = ::onDismiss,
                         grantNotificationPermission = {
                             ensureMarketingChannelExists(context.applicationContext)
+                            requireComponents.notificationsDelegate.requestNotificationPermission()
```

Comment:
Depending on the experiment results we could now better handle a potential rejection and show an additional message (even by just changing the text) before we ask for the permission a second time.